File: C++ Code/Basics/Recursion/Power.cpp
// Q. WAP to print the value of 'a' raised to power 'b' using recursion. a and b are given by user.
#include<iostream>
using namespace std;
int power(int a, int b){
    if(b==0)
        return 1;
    if(b%2==0)
        return power(a,b/2)*power(a,b/2);
    if(b%2!=0)
        return a*power(a,b/2)*power(a,b/2);
}
int main(){
    int a,b;
    cout<<"Enter base, a = ";
    cin>>a;
    cout<<"Enter exponent, b = ";
    cin>>b;
    cout<<a<<" raised to the power "<<b<<" = "<<power(a,b);
    return 0;
}

File: Java Code/Basics/Recursion/Power.java
import java.util.Scanner;

public class Power{
  public static float pow(float x, int n){
    if(n==0)
      return 1;
    if(n==1)
      return x;
    return x*pow(x,n-1);
  }
  public static void main(String[] args){
    Scanner sc=new Scanner(System.in);
    System.out.print("Enter the base : ");
    float x=sc.nextFloat();
    System.out.print("Enter the exponent : ");
    int n=sc.nextInt();
    boolean isNegative=false;
    if(n<0)
      isNegative=true;
    n=Math.abs(n);
    if(isNegative)
      System.out.println(x+" raised to power "+n+" = "+1/pow(x,n));
    else
      System.out.println(x+" raised to power "+n+" = "+pow(x,n));
  }
}